src/replicate/getDocs.js

```diff
@@ -1,10 +1,49 @@
 import clone from '../deps/clone';
+import flatten from '../deps/flatten';
 import Promise from '../deps/promise';
 
 function isGenOne(rev) {
   return /^1-/.test(rev);
 }
 
+function fileHasChanged(localDoc, remoteDoc, filename) {
+  return !localDoc._attachments ||
+         !localDoc._attachments[filename] ||
+         localDoc._attachments[filename].digest !== remoteDoc._attachments[filename].digest
+}
+
+function getDocAttachments(db, doc) {
+  var filenames = Object.keys(doc._attachments)
+  return Promise.all(filenames.map(function (filename) {
+    return db.getAttachment(doc._id, filename);
```

Comment:
We could also check all non-leaf non-deleted revisions here, but for now I think this is fine. That might be overkill anyway.